packages/next/client/image.tsx

```diff
@@ -119,6 +125,7 @@ type GenImgAttrsData = {
   width?: number
   quality?: number
   sizes?: string
+  loader?: URLResolver
```

Comment:
```suggestion
  loader?: URLResolver | LoaderValue
```

Anyway we could make this work? It would be nice to avoid `next.config.js` completely but we would also need `path` 🤔

Reply:
So you're thinking it should be possible to do `<Image src="foo.jpg" loader="imgix" root="https://mydomain.com" />` ? Or would we use the root defined in the `next.config.js`?

Reply:
Yeah I think that could work.

Or maybe we should just export the loaders instead so you could do:

```jsx
import Image, {imgixLoader} from 'next/image'
<Image src="foo.jpg" loader={imgixLoader} />
```

I think I like that better. What do you think @Timer?

Reply:
The root would have to get into the loader at some point. So it could be `<Image src="foo.jpg" loader={imgixLoader('https://mydomain.com')} />` maybe? Would this all be on top of the current functionality, for backward compatibility?